src/core/s.rs

```diff
@@ -33,6 +33,16 @@ impl S {
         h.extend(b.0.to_vec());
         Self(*hash(&h).as_bytes())
     }
+
+    pub fn xor(a: Self, b: Self) -> Self {
+        Self(
+            zip(a.0, b.0)
+                .map(|(u, v)| u ^ v)
+                .collect::<Vec<u8>>()
+                .try_into()
+                .unwrap(),
+        )
+    }
```

Comment:
For arrays, this is a very expensive method. Since we know the size of arrays and it is a constant, a loop with an explicit indication of the size is much more optimal.

Also you take ownership of both values, you can take the memory of one of them and reuse it

Reply:
https://godbolt.org/z/qETnWsja5